src/main/java/net/dv8tion/jda/api/interactions/components/ActionComponent.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.interactions.components;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Component which supports interactions via {@link ComponentInteraction}.
+ */
+public interface ActionComponent extends Component
```

Comment:
That would make all the other getters very ugly with `List<ActionComponent<?>>`

Reply:
And they're normally just `List<ActionComponent<?>>`?
I don't really see much of a difference there tbh, but I guess that it would mean that _users_ would have to deal with return types having `?` in them which is non ideal... 🤔

Reply:
I would prefer not giving users wildcard types if possible

Reply:
Yeah. Right now there are very few places where Wildcard are exposed to users and that's for the best.

Sounds good. We just need to make sure that we always override to provide the more-specific stuff on method returns. It's very easy to forget.